### DISCUSSION TITLE
UDM Pro 3.x - nspawn Pihole - How to use Pihole as DHCP server

### DISCUSSION THREAD

**Discussion by Antergosgeek** [unifi-utilities/unifi-common#616] [Q&A] [answered]
Hello,

I followed the instructions to install a debian-custom nspawn container and installed pihole on it:
https://github.com/unifi-utilities/unifios-utilities/tree/main/nspawn-container
https://github.com/unifi-utilities/unifios-utilities/blob/main/nspawn-container/examples/pihole/README.md

Does anyone know how to use Pihole as the DHCP server though so that I can get the actual client IP instead of just 10.0.5.1?

**Comment by Antergosgeek** [comment]
Anyone?

**Reply by m4n63** [reply, marked as answer]
I have not used PiHole as a replacement DHCP server myself, but I would assume you would just enable the dhcp server in your PiHole and create the scopes you want to have there, and then in the UDM enabe the DHCP relay on the relevant networks and point the relay to the IP of the PiHole.

**Reply by Antergosgeek** [reply]
I’ll try that, thank you!
